src/AddressSpace.cc

```diff
@@ -1363,6 +1364,7 @@ void AddressSpace::coalesce_around(MemoryMap::iterator it) {
       break;
     }
     assert(first_kv->second.emu_file == next->second.emu_file);
+    assert(first_kv->second.local_addr == next->second.local_addr);
```

Comment:
I assume that `local_addr` should point to the first byte of the map, otherwise we won't be able to use it effectively, in which case this assertion should *not* hold. Instead we'll need to assert that `first_kv->second.local_addr + size == next->second.local_addr` ... and we'll need to update local_addr when splitting maps so that this assertion holds.